example/main.js

```diff
@@ -0,0 +1,134 @@
+const ENSImplementation = require("../build/contracts/ensimplementation.json");
+const DNSRegistrar = require("../build/contracts/dnsregistrar.json");
+const Web3      = require('web3');
+const dns = require("@ensdomains/dnssec-oracle/lib/dns.js");
+
+function updateDOM(element, message, override){
+  if(override){
+    document.getElementById(element).innerHTML = message;
+  }else{
+    document.getElementById(element).innerHTML = document.getElementById(element).innerHTML   + '\n' + message;
+  }
+}
+
+function askOracle(){
+  document.getElementById("oracle-output").innerHTML = '';
+  let proofs = window.result.proofs;
+  proofs.forEach(function(proof){
+    let rrdata;
+    window.oracle.knownProof(proof).then(function(r){
+      updateDOM('oracle-output', proof.name + '\t' + proof.type + '\t' + r)
+    });
+  })
+  updateDOM('oracle-output', 'The address is owned by ' +  window.result.owner)
+}
+
+function askEns(input, cb){
+  console.log('owner', input, namehash.hash(input));
+  window.ens.owner(namehash.hash(input), (error, r)=>{
+    console.log('ens', error, r);
+    cb(r);
+  })
+}
+
+function claim(name, proof){
+  let encodedProof = '0x' + proof.rrdata.toString('hex');
+  window.dnsregistrar.claim(dns.hexEncodeName(name + '.'), encodedProof, {from: web3.eth.defaultAccount}, (error, r)=>{
+    console.log('claimed', r);
+  });
+}
+
+function submitProof(proofs, i){
+  let proof = proofs[i]
+  window.oracle.knownProof(proof).then(function(r){
+    if(r == '0x0000000000000000000000000000000000000000'){
+      window.oracle.submitProof(proof, proofs[i-1], {from: web3.eth.defaultAccount}).then(function(r){
+        console.log('result', i, r);
+        if(i < proofs.length - 1){
+          submitProof(proofs, i+1)
+        }else{
+          askEns(window.input, (r)=>{
+            if(r == '0x0000000000000000000000000000000000000000'){
+              claim(window.input, proof)
+            }
+          })  
+          updateDOM('oracle-output', 'Click Lookup button to check the latest state');
+        }
+      });
+    }else{
+      if(i < proofs.length - 1){
+        submitProof(proofs, i+1)
+      }else{
+        askEns(window.input, (r)=>{
+          if(r == '0x0000000000000000000000000000000000000000'){
+            claim(window.input, proof)
+          }
+        })
+        updateDOM('oracle-output', 'Click Lookup button to check the latest state');
+      }
+    }
+  });
+}
+
+document.addEventListener("DOMContentLoaded", function(event) {
+  if (typeof web3 !== 'undefined') {
+    // Use the browser's ethereum provider
+    var provider = web3.currentProvider
+    console.log('Using metamask')
+  } else {
+    var provider = new Web3.providers.HttpProvider("http://localhost:8545");
+    console.log('Using local provider')
+  }
+
+  // They are pre web3 1.0 syntax loaded via metamask
+  web3.version.getNetwork((error, network)=>{
+    const oracleAddress = DNSSEC.networks[network].address;
```

Comment:
Doesn't Truffle provide `DNSSEC.instance()` to avoid needing to do all this?

Reply:
My initial intention of this example is to use it without relying on truffle based project. In reality `require("../build/contracts/DNSSEC.json");` assumes that the project resides within truffle but I can replace it with `require("@ensdomains/dnssec-oracle/build/contracts/DNSSEC.json")` which is automatically included when user does `npm install`. The alternative idea is to move this example file to `dnsregistrar` so that users can manually try out end to end from proving to claiming the name to ens.